packages/docusaurus-theme-classic/src/theme/Toggle/index.tsx

```diff
@@ -31,21 +30,63 @@ const Light = ({icon, style}: IconProps): JSX.Element => (
 );
 
 export default function (props: Props): JSX.Element {
+  const {className, checked: defaultChecked, onChange} = props;
   const {
     colorMode: {
       switchConfig: {darkIcon, darkIconStyle, lightIcon, lightIconStyle},
     },
   } = useThemeConfig();
   const {isClient} = useDocusaurusContext();
+  const [checked, setChecked] = useState(defaultChecked);
```

Comment:
Can you extract all the non-docusaurus code to a separate function and mention in a comment it's inspired from react-toggle?

Also, the original comp used PureComponent so we can also use React.memo() around that newly extracted component too (the onChange is already memorized)

Reply:
It seems redundant to me to create another component as like before, I also don't understand the gist of memoizing this new child component, but okay I'll do it.

Reply:
I get it, it's just that our earlier components didn't follow that pattern, although we'll need to refactor them the same way, I guess.

Reply:
yes that's the goal but no hurry for now